js/data/bucket.js

```diff
@@ -223,7 +225,7 @@ Bucket.prototype.setAttribPointers = function(programName, gl, program, offset,
     var disabledAttributes = this.attributes[programName].disabled;
     for (var i = 0; i < disabledAttributes.length; i++) {
         attribute = disabledAttributes[i];
-        if (attribute.isLayerConstant === false && layer.id !== attribute.layerId) continue;
+        assert(attribute.paintProperty);
```

Comment:
We need to check `layer.id !== attribute.layerId` to handle the case where there are multiple layers using property functions (data-driven styling functions) sharing a `ref` parent.